airflow/providers/databricks/plugins/databricks_workflow.py

```diff
@@ -0,0 +1,440 @@
+            raise AirflowException("No launch task can be found in the task group.") from e
+        launch_task_id = get_launch_task_id(task_group.parent_group)
+
+    return launch_task_id
+
+
+def _get_launch_task_key(current_task_key: TaskInstanceKey, task_id: str) -> TaskInstanceKey:
+    """
+    Return the task key for the launch task.
+
+    This allows us to gather databricks Metadata even if the current task has failed (since tasks only
+    create xcom values if they succeed).
+
+    :param current_task_key: The task key for the current task.
+    :param task_id: The task ID for the current task.
+    :return: The task key for the launch task.
+    """
+    if task_id:
+        return TaskInstanceKey(
+            dag_id=current_task_key.dag_id,
+            task_id=task_id,
+            run_id=current_task_key.run_id,
+            try_number=current_task_key.try_number,
+        )
+    else:
+        return current_task_key
+
+
+@provide_session
+def get_task_instance(operator, dttm, session: Session = NEW_SESSION):
+    dag_id = operator.dag.dag_id
+    dag_run = DagRun.find(dag_id, execution_date=dttm)[0]
+    ti = (
+        session.query(TaskInstance)
+        .filter(
+            TaskInstance.dag_id == dag_id,
+            TaskInstance.run_id == dag_run.run_id,
+            TaskInstance.task_id == operator.task_id,
+        )
+        .one_or_none()
+    )
+    if not ti:
+        raise TaskInstanceNotFound("Task instance not found")
+    return ti
+
+
+def get_xcom_result(
+    ti_key: TaskInstanceKey,
+    key: str,
+    ti: TaskInstance | None,
+) -> Any:
+    result = XCom.get_value(
+        ti_key=ti_key,
+        key=key,
+    )
+    from airflow.providers.databricks.operators.databricks_workflow import WorkflowRunMetadata
+
+    return WorkflowRunMetadata(**result)
+
+
+class WorkflowJobRunLink(BaseOperatorLink, LoggingMixin):
+    """Constructs a link to monitor a Databricks Job Run."""
+
+    name = "See Databricks Job Run"
+
+    def get_link(
+        self,
+        operator: BaseOperator,
+        dttm=None,
+        *,
+        ti_key: TaskInstanceKey | None = None,
+    ) -> str:
+        ti = None
+        if not ti_key:
+            ti = get_task_instance(operator, dttm)
+            ti_key = ti.key
+        task_group = operator.task_group
+
+        if not task_group:
+            raise AirflowException("Task group is required for generating Databricks Workflow Job Run Link.")
+
+        dag = airflow_app.dag_bag.get_dag(ti_key.dag_id)
+        dag.get_task(ti_key.task_id)
+        self.log.info("Getting link for task %s", ti_key.task_id)
+        if ".launch" not in ti_key.task_id:
+            self.log.debug("Finding the launch task for job run metadata %s", ti_key.task_id)
+            launch_task_id = get_launch_task_id(task_group)
+            ti_key = _get_launch_task_key(ti_key, task_id=launch_task_id)
+        # Should we catch the exception here if there is no return value?
```

Comment:
Probably not? Even if we catch here, error will still occur when we build the url

Reply:
or maybe we can return " when an exception is raised.

Reply:
yes, I think would be wise revisit this later after.